### DISCUSSION TITLE
Where to find info about editorCallBack ?

### DISCUSSION THREAD

**Discussion by webdevaw** [primefaces/community#717] [PrimeReact] [answered]
Hi,
my question is where I can find some information about **editorCallback**?

In **DataTable** when adding editing option in order to update the typed value immediately, you need to call the **options.editorCallback** function in the **InputText onChange** event. 
This updates the **options.value** and renders the updated value.

Like this:
`<InputText type="text" value={options.value} onChange={e => options.editorCallback(e.target.value) />`

I did that and it works fine but I can't find anything on PrimeReact website about **editorCallback** (except it is used in code example related to editing functionality). 
But there is no any info on it, what it is, how to use it, how it works etc.
All I found on internet is one answer on stackoverflow from Jun 2 2022 just explaining how to use it.
I'm new to PrimeReact and Reactjs in general and I just try to understand how things works.

**Comment by melloware** [comment, marked as answer]
You are right its not on the docs page but here it is declared in the TypeScript.

```ts
interface ColumnEditorOptions {
    /**
     * Node element of the editor.
     */
    node?: any;
    /**
     * Data of the edited row.
     */
    rowData: any;
    /**
     * Value of the editor.
     */
    value: any;
    /**
     * Column of the editor.
     */
    column: Column;
    /**
     * Field name of the column.
     */
    field: string;
    /**
     * Index of the edited row.
     */
    rowIndex: number;
    /**
     * Whether the row is frozen or not.
     */
    frozenRow?: boolean;
    /**
     * Additional properties passed to the body component.
     */
    props: any;
    /**
     * Callback function triggered on editor change.
     * @param {*} value - Browser event
     */
    editorCallback?(value: any): void;
}
```
